salt/modules/mongodb.py

```diff
@@ -297,12 +297,14 @@ def user_create(
     if not roles:
         roles = []
 
+    _roles = [{"role": _role, "db": database} for _role in roles]
     try:
         log.info("Creating user %s", name)
         mdb = pymongo.database.Database(conn, database)
-        mdb.add_user(name, passwd, roles=roles)
+        ret = mdb.command("createUser", name, pwd=passwd, roles=_roles)
```

Comment:
why are you adding this call to a variable, when the variables is not being used.

This also applies to line 331 and 406

Reply:
Something left over from testing.